common/djangoapps/track/tests/test_middleware.py

```diff
@@ -102,14 +102,14 @@ def test_single_forward_for_header_ip_context(self):
 
     def test_multiple_forward_for_header_ip_context(self):
         request = self.request_factory.get('/courses/')
-        remote_addr = '127.0.0.1'
-        forwarded_ip = '11.22.33.44, 10.0.0.1, 127.0.0.1'
+        remote_addr = '10.0.0.1'  # reverse proxy, e.g. nginx
+        forwarded_ip = '11.22.33.44, 5.6.7.8'  # what nginx sent us
```

Comment:
1. Do to importance of this code, I think we should have coverage to changes to the settings, and fir the variety of conditions and logging in the code.
2. Consider assertion on the setting defaults in this test.

Reply:
Can you say more about this? Do you mean specifically for the track djangoapp, or for the new IP code in general?

Reply:
New IP code in general. I wasn’t clear on the file context when I wrote this.